libraries/wallet/wallet.cpp

```diff
@@ -758,6 +758,56 @@ class wallet_api_impl
 
       return true;
    }
+
+   signed_transaction multisig_sign_transaction( signed_transaction tx,
+                                                 const vector<string> &wif_keys, bool broadcast )
+   {
+      vector<fc::ecc::private_key> private_keys;
+
+      for ( const string &wif_key : wif_keys )
+      {
+         fc::optional<fc::ecc::private_key> optional_private_key = wif_to_key( wif_key );
+
+         if ( !optional_private_key )
+         {
+            elog( "Invalid private key: ${i}", ( "i", wif_key ) );
+            FC_THROW("");
+         }
+
+         private_keys.push_back( *optional_private_key );
+      }
+
+      for ( const fc::ecc::private_key &privkey : private_keys )
+      {
+         signature_type sig = privkey.sign_compact( tx.sig_digest( _chain_id ) );
+         const vector<signature_type> &sigs = tx.signatures;
+         if ( std::find( sigs.begin(), sigs.end(), sig ) != sigs.end() )
+         {
+            wlog( "signature of ${fn} had already exist, skipping", ( "fn", key_to_wif( privkey ) ) );
```

Comment:
Please do not log private keys. Logging the corresponding public key should be sufficient.